src/mut/morphology.cpp

```diff
@@ -163,6 +164,23 @@ void Morphology::eraseByValue(std::vector<std::shared_ptr<Section>>& vec,
     vec.erase(std::remove(vec.begin(), vec.end(), section), vec.end());
 }
 
+
+void inline insertSectionsBeforeSection(std::vector<std::shared_ptr<Section>>& sections_to_update,
+                                        const std::vector<std::shared_ptr<Section>>& sections,
+                                        uint32_t section_id) {
+    // lambda to check if a section has the given id
+    auto equals_section_id = [section_id](const std::shared_ptr<Section>& section) {
+        return section->id() == section_id;
```

Comment:
for immut; the using the `section_id` makes sense, I think, but what about for mut morphs that have been "built up" in a non-depth first fashion - is that possible, and does this operation work in that context?

Reply:
Adding a thorough test for the different traversals indicates that bfs works correctly after all. :)